packages/entities/entities-plugins/src/components/free-form/shared/types.ts

```diff
@@ -119,3 +119,45 @@ export type PartialNotification = {
  * @deprecated use `useToaster` instead
  */
 export type GlobalAction = 'notify'
+
+/**
+ * Rules to control the rendering of form fields.
+ * Only `Form` and `ObjectField` components can accept these rules
+ */
+export interface RenderRules {
+  /**
+   * Bundles of fields to be rendered together.
+   * - Each bundle is an array of field paths.
+   * - Each bundle should be in the same level.
+   * - Circular references between bundles are not allowed.
+   * @example
+   * ```ts
+   * [
+   *   ['config.username', 'config.password'], // first bundle
+   *   ['config.strategy', 'config.redis'], // second bundle
+   *   ['config.strategy', 'config.cache.redis'], // ❌ different levels are not allowed
+   *   ['config.redis', 'config.strategy'], // ❌ circular reference not allowed
+   * ]
+   */
+  bundles?: string[][]
+
+  /**
+   * Dependencies between fields to control their visibility.
+   * - A field will be shown only if its dependency is satisfied.
+   * - A field will only have a value if its dependency is satisfied.
+   * - The `fieldValue` will be deeply compared.
+   * - Field and its dependency should be in the same level.
+   * - Circular dependencies are not allowed.
+   * @example
+   * ```ts
+   * {
+   *   'config.redis': ['config.strategy', 'redis'],
```

Comment:
Question:
1. do we support showing `config.a` only when `config.b` value is one of `foo` and `bar`?
2. do we support showing `config.a` only when `config.b` value is `foo` and `config.c` value is`bar`?